sisl/viz/plotly/gui/api.py

```diff
@@ -0,0 +1,184 @@
+import shutil
+import pathlib
+import traceback
+
+from plotly.graph_objects import Figure
+import numpy as np
+import simplejson
+from simplejson.encoder import JSONEncoder
+from functools import partial
+
+import flask
+from flask import Flask, request, jsonify, make_response
+from flask_socketio import SocketIO, join_room
+
+from sisl.viz import BlankSession, Plot
+from ..plotutils import load
+from .api_utils import with_user_management, if_user_can, listen_to_users, \
+    emit_plot, emit_session, emit_error, emit_loading_plot, emit
+
+__DEBUG = False
+
+app = Flask("SISL GUI API")
```

Comment:
We have to remove the implicit call to `Flask`.

My guess is that this will be relatively slow to call python scripts with `sisl` imports just because `Flask` creates stuff on every invocation.

Some way of using `create_app` or whatever would suffice.

Reply:
That is right, I have put the creation of the app into a function that is only called when the GUI is launched.